testing.md

```diff
@@ -0,0 +1,299 @@
+did it.
+
+Example:
+
+Just for fun let's also make our test record the performance timeline. Let's
+create a test file `my_app/test_driver/user_list_scrolling_test.dart`:
+
+```dart
+import 'dart:async';
+
+// Imports the Flutter Driver API
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('scrolling performance test', () {
+    FlutterDriver driver;
+
+    setUpAll(() async {
+      // Connects to the application
+      driver = await FlutterDriver.connect();
+    });
+
+    tearDownAll(() async {
+      if (driver != null) {
+        // Closes the connection
+        driver.close();
+      }
+    });
+
+    test('measure', () async {
+      // Record the performance timeline of things that happen inside the closure
+      Timeline timeline = await driver.traceAction(() async {
+        // Find the scrollable user list
+        ObjectRef userList = await driver.findByValueKey('user-list');
+        expect(userList, isNotNull);
+
+        // Scroll down 5 times
+        for (int i = 0; i < 5; i++) {
+          // Scroll 300 pixels down, for 300 millis
+          await driver.scroll(userList, 0.0, -300.0, new Duration(milliseconds: 300));
+
+          // Emulate a user's finger taking its time to go back to the original
+          // position before the next scroll
+          await new Future<Null>.delayed(new Duration(milliseconds: 500));
+        }
+
+        // Scroll up 5 times
+        for (int i = 0; i < 5; i++) {
+          await driver.scroll(userList, 0.0, 300.0, new Duration(milliseconds: 300));
+          await new Future<Null>.delayed(new Duration(milliseconds: 500));
+        }
+      });
+
+      // The `timeline` object contains all the performance data recorded during
+      // the scrolling session. It can be digested into a handful of useful
+      // aggregate numbers, such as "average frame build time".
+      TimelineSummary summary = new TimelineSummary.summarize(timeline);
+      summary.writeSummaryToFile('stocks_scroll_perf', pretty: true);
+      summary.writeTimelineToFile('stocks_scroll_perf', pretty: true);
+    });
+  });
+}
+```
+
+### Run test
+
+To run the test on an Android device, connect the device via USB to your
+computer and enable USB debugging. Then run the following command:
+
+```
+flutter drive --target=my_app/test_driver/user_list_scrolling.dart
+```
+
+This command will:
+
+* build the `--target` application and install it on the device
+* launch the application
+* run the `my_app/test_driver/user_list_scrolling_test.dart` test
+
+You might be wondering how the command finds the correct test file. The
+`flutter drive` command uses a convention to look for the test file in the same
+directory as the instrumented `--target` application that has the same file name
+but for the `_test` suffix in it.
+
+**NOTE: We currently only support Android devices and iOS Simulator. Support for
```

Comment:
Please move this note further above, right after where you first describe the Flutter driver.